src/main/scala/org/ergoplatform/modifiers/history/NipopowProofModifier.scala

```diff
@@ -0,0 +1,41 @@
+package org.ergoplatform.modifiers.history
+
+import org.ergoplatform.modifiers.ErgoPersistentModifier
+import org.ergoplatform.modifiers.history.popow.NipopowProof
+import org.ergoplatform.settings.Algos
+import scorex.core.ModifierTypeId
+import scorex.core.serialization.ScorexSerializer
+import scorex.util.{ModifierId, bytesToId}
+
+/**
+  * Network message carrying NiPoPoW proof for a block.
+  *
+  * Not really implemented yet. Will be used for bootstrapping.
+  *
+  * @param proof   - NiPoPoW proof
+  * @param sizeOpt - optionally, serialized network message size
+  */
+case class NipopowProofModifier(proof: NipopowProof, override val sizeOpt: Option[Int] = None)
+  extends Comparable[NipopowProofModifier] with Ordered[NipopowProofModifier] with ErgoPersistentModifier {
+
+  override val modifierTypeId: ModifierTypeId = NipopowProofModifier.modifierTypeId
+
+  override def parentId: ModifierId = ???
```

Comment:
Should this really be ErgoPersistentModifier. It looks unsound.

Reply:
for now not, but the plan is to send it over p2p network and use for blockchain bootstrapping